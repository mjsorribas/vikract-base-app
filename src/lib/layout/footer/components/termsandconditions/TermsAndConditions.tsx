import { Box, Stack, Link } from "@chakra-ui/react";

export default function TermsAndConditions() {
  return (
    <Stack align={{ md: "flex-end", sm: "center" }} spacing={6}>
      <Box fontSize="sm">
        <Link href="https://vikract.com/terms" ml={3}>
          Terms
        </Link>
        <Link href="https://vikract.com/conditions" ml={3}>
          Conditions
        </Link>
        <Link href="https://vikract.com/cookies" ml={3}>
          Cookies
        </Link>
      </Box>
    </Stack>
  );
}
