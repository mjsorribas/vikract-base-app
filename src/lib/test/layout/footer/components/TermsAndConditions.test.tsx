import { ChakraProvider } from "@chakra-ui/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test, vi } from "vitest";

// Mock de una dependencia específica
vi.mock("@chakra-ui/react", async () => {
  const actual = await vi.importActual("@chakra-ui/react");
  return {
    ...actual,
    useBreakpointValue: () => "mockedValue",
    extendTheme: actual.extendTheme, // Asegúrate de incluir extendTheme en el mock
  };
});

import TermsAndConditions from "lib/layout/footer/components/termsandconditions/TermsAndConditions";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("TermsAndConditions", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <ChakraProvider>
        <TermsAndConditions />
      </ChakraProvider>
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
