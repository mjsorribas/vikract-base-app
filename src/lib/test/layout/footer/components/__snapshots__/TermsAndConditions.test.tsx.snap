// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TermsAndConditions 1`] = `
<div
  className="chakra-stack css-14dnb91"
>
  <div
    className="css-g5hezf"
  >
    <a
      className="chakra-link css-9n3bbc"
      href="https://vikract.com/termsandconditions"
    >
      Terms
    </a>
    <a
      className="chakra-link css-9n3bbc"
      href="https://vikract.com/termsandconditions"
    >
      Conditions
    </a>
    <a
      className="chakra-link css-9n3bbc"
      href="https://vikract.com/cookies"
    >
      Cookies
    </a>
  </div>
</div>
`;
